presto-clickhouse/src/main/java/io/prestosql/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,472 @@
+    {
+        List<JdbcColumnHandle> columns = new ArrayList<>();
+        try {
+            ClickHouseConnection connection = connectionFactory.openConnection(session)
+                    .unwrap(ClickHouseConnection.class);
+            Map<String, Integer> arrayColumnDimensions = ImmutableMap.of();
+            ResultSet resultSet = connection.getMetaData().getColumns(null, tableHandle.getSchemaName(),
+                    tableHandle.getTableName(), null);
+            while (resultSet.next()) {
+                String columnName = resultSet.getString("COLUMN_NAME");
+                JdbcTypeHandle typeHandle = new JdbcTypeHandle(resultSet.getInt("DATA_TYPE"),
+                        Optional.of(resultSet.getString("TYPE_NAME")), resultSet.getInt("COLUMN_SIZE"),
+                        resultSet.getInt("DECIMAL_DIGITS"), Optional.ofNullable(arrayColumnDimensions.get(columnName)));
+                Optional<ColumnMapping> columnMapping = toPrestoType(session, connection, typeHandle);
+                log.debug("Mapping data type of '%s' column '%s': %s mapped to %s", tableHandle.getSchemaTableName(),
+                        columnName, typeHandle, columnMapping);
+                // skip unsupported column types
+                if (columnMapping.isPresent()) {
+                    boolean nullable = (resultSet.getInt("NULLABLE") != columnNoNulls);
+                    Optional<String> comment = Optional.ofNullable(resultSet.getString("REMARKS"));
+                    columns.add(JdbcColumnHandle.builder().setColumnName(columnName).setJdbcTypeHandle(typeHandle)
+                            .setColumnType(columnMapping.get().getType()).setNullable(nullable).setComment(comment)
+                            .build());
+                }
+                if (columnMapping.isEmpty()) {
+                    UnsupportedTypeHandling unsupportedTypeHandling = getUnsupportedTypeHandling(session);
+                    verify(unsupportedTypeHandling == IGNORE,
+                            "Unsupported type handling is set to %s, but toPrestoType() returned empty",
+                            unsupportedTypeHandling);
+                }
+            }
+            if (columns.isEmpty()) {
+                // In rare cases a table might have no columns.
+                throw new TableNotFoundException(tableHandle.getSchemaTableName());
+            }
+            return ImmutableList.copyOf(columns);
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    public WriteMapping toWriteMapping(ConnectorSession session, Type type)
+    {
+        if (BOOLEAN.equals(type)) {
+            return WriteMapping.booleanMapping("UInt8", booleanWriteFunction());
+        }
+        if (REAL.equals(type)) {
+            return WriteMapping.longMapping("Float32", realWriteFunction());
+        }
+        if (TIMESTAMP_MILLIS.equals(type)) {
```

Comment:
can you please sort the mappings:

- boolean
- tinyint, 
- smallint, 
- integer, 
- bigint
- real, 
- double
- decimal
- char, 
- varchar, 
- varbinary
- date
- time, 
- time tz
- timestamp 
- timestamp tz

?

This will help eye-ball coverage better and generally will make reading the code easier.

(of curse, no need to add the mappings that are not supported yet)

Reply:
(same for `toPrestoType` method)

Reply:
ok